spec/unit/coinbase/staking_reward_spec.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+describe Coinbase::StakingReward do
+  let(:network_id) { :network_id }
+  let(:asset_id) { :asset_id }
+  let(:asset) { instance_double(Coinbase::Asset) }
+  let(:start_time) { Time.now }
+  let(:end_time) { Time.now }
+  let(:address_ids) { %w[address_id] }
+  let(:stake_api) { instance_double(Coinbase::Client::StakeApi) }
+  let(:format) { :usd }
+  let(:has_more) { false }
+  let(:staking_reward_model) { instance_double(Coinbase::Client::StakingReward, amount: 100) }
+
+  before do
+    allow(Coinbase::Asset).to receive(:fetch).and_return(asset)
+    allow(Coinbase::Client::StakeApi).to receive(:new).and_return(stake_api)
+    allow(stake_api).to receive(:fetch_staking_rewards).and_return(
+      instance_double(Coinbase::Client::FetchStakingRewards200Response, data: [staking_reward_model], has_more: true),
+      instance_double(Coinbase::Client::FetchStakingRewards200Response, data: [], has_more: false)
+    )
+  end
+
+  describe '.list' do
+    subject(:list) do
+      described_class.list(
+        network_id, asset_id,
+        address_ids,
+        start_time: start_time,
+        end_time: end_time,
+        format: format
+      )
+    end
+
+    it 'loads the asset' do
+      list.to_a
+
+      expect(Coinbase::Asset).to have_received(:fetch).with(network_id, asset_id)
+    end
+
+    it 'fetches the staking rewards' do
+      list.to_a
+
+      expect(stake_api).to have_received(:fetch_staking_rewards).with(
+        network_id: 'network-id',
+        asset_id: asset_id,
+        address_ids: address_ids,
+        start_time: start_time.iso8601,
+        end_time: end_time.iso8601,
+        format: format,
+        next_page: nil
```

Comment:
I would expect this to be called once with `next_page` set and once without it set?